apps/tester-federation-v2/rspack.config.host-app.mjs

```diff
@@ -0,0 +1,167 @@
+    throw new Error('Missing platform');
+  }
+
+  process.env.BABEL_ENV = mode;
+
+  return {
+    mode,
+    devtool: false,
+    context,
+    entry,
+    resolve: {
+      ...Repack.getResolveOptions(platform),
+    },
+    output: {
+      clean: true,
+      hashFunction: 'xxhash64',
+      path: path.join(dirname, 'build', 'host-app', platform),
+      filename: 'index.bundle',
+      chunkFilename: '[name].chunk.bundle',
+      publicPath: Repack.getPublicPath({ platform, devServer }),
+      uniqueName: 'MF2-HostApp',
+    },
+    optimization: {
+      minimize,
+      chunkIds: 'named',
+    },
+    module: {
+      rules: [
+        Repack.REACT_NATIVE_LOADING_RULES,
+        Repack.NODE_MODULES_LOADING_RULES,
+        /* repack is symlinked to a local workspace */
+        {
+          test: /\.[jt]sx?$/,
+          type: 'javascript/auto',
+          include: [/repack[/\\]dist/],
+          use: {
+            loader: 'builtin:swc-loader',
+            options: {
+              env: { targets: { 'react-native': '0.74' } },
+              jsc: { externalHelpers: true },
+            },
+          },
+        },
+        /* codebase rules */
+        {
+          test: /\.[jt]sx?$/,
+          type: 'javascript/auto',
+          exclude: [/node_modules/, /repack[/\\]dist/],
+          use: {
+            loader: 'builtin:swc-loader',
+            /** @type {import('@rspack/core').SwcLoaderOptions} */
+            options: {
+              sourceMaps: true,
+              env: {
+                targets: { 'react-native': '0.74' },
+              },
+              jsc: {
+                externalHelpers: true,
+                transform: {
+                  react: {
+                    runtime: 'automatic',
+                  },
+                },
+              },
+            },
+          },
+        },
+        Repack.REACT_NATIVE_CODEGEN_RULES,
+        {
+          test: Repack.getAssetExtensionsRegExp(Repack.ASSET_EXTENSIONS),
+          use: {
+            loader: '@callstack/repack/assets-loader',
+            options: {
+              platform,
+              devServerEnabled: Boolean(devServer),
+            },
+          },
+        },
+      ],
+    },
+    plugins: [
+      new Repack.RepackPlugin({
+        context,
+        mode,
+        platform,
+        devServer,
+        output: {
+          bundleFilename,
+          sourceMapFilename,
+          assetsPath,
+        },
+      }),
+      new Repack.plugins.ModuleFederationPlugin({
+        name: 'HostApp',
+        filename: 'HostApp.container.js.bundle',
+        remotes: {
+          MiniApp: `MiniApp@http://localhost:8082/${platform}/mf-manifest.json`,
+        },
+        dts: false,
+        runtimePlugins: [
+          resolve('@callstack/repack/federation-runtime-plugin'),
+          path.resolve(dirname, 'src', 'utils', 'runtime-debug.ts'),
+        ],
+        shared: {
+          react: {
+            singleton: true,
+            eager: true,
```

Comment:
Why do they have to be eager?

Reply:
`react-native` and `react` need to be present for initialisation of the environment done in `react-native/Libraries/Core/InitializeCore.js` - Hermes engine doesn't have event loop on its own -> no microtasks -> no timers like `setImmediate` or `setTimeout`.

Reply:
So I'm guessing there's no promise either? Like wouldn't it help if the entrypoint itself returned a promise so that entry setup itself is asynchronous? Like my hoisted runtime experiment does.

Reply:
I wonder if lynx supports event loops. 🤔 
This is something I'll probably want to bake into our vm if possible

Reply:
> So I'm guessing there's no promise either? Like wouldn't it help if the entrypoint itself returned a promise so that entry setup itself is asynchronous? Like my hoisted runtime experiment does.

yeah that would be ideal, but that would need more tinkering. I think I tried with hoisted runtime once but didn't pursue it much further. The biggest limiting factor is maintenance of custom initialisation logic. 

On the other side, @robik once had an idea of creating an extra layer between RN Core and Hermes called RN Runtime which could be a place for stuff like this. Not sure whats the status of that, maybe he can share something here as well :)

Reply:
Yeah, I think this would solve this issue. The RFC is in pre-published state for now, should be up for discussions soon :)